examples/resources/awscc_chatbot_slack_channel_configuration/first_example.tf

```diff
@@ -0,0 +1,14 @@
+
+resource "awscc_chatbot_slack_channel_configuration" "example" {
+  # Set the name to identify the configuration in the ChatBot console
+  configuration_name = "ChatBot-Example"
+
+  # Replace <account_id> and <role-name> with sutable values
+  iam_role_arn = "arn:aws:iam::<account_id>:role/<role-name>"
```

Comment:
what would be a good example of IAM roles and permissions that required for this chatbot? can we add that as part of the example?

Reply:
There is an AWS Managed role that can be used [AWSServiceRoleForAWSChatbot](https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-west-2#/roles/details/AWSServiceRoleForAWSChatbot)

Should I just set the arn to that and add a made up account number?

Reply:
you can use the interpolation like this:

```
 arn:${data.aws_partition.current_partition.id}:iam::${data.aws_caller_identity.current_account.account_id}:root"
```

or you could also use data-source to find the role and then reference it
```
data "aws_iam_role" "example" {
  name = "an_example_role_name"
}
```

Reply:
From reading the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-chatbot-slackchannelconfiguration.html), this IAM role defines the permission for AWS Chatbot and not the service-linked role.

I believe the right step would be to create the IAM role, permissions and trust policy and then refer it in the `awscc_chatbot_slack_channel_configuration`